deprecate `copy_arrays` with warning (#1797)
asdf-format/asdf: README.rst

    with asdf.open("example.asdf") as af:
        ...

.. warning::
    The ``copy_arrays`` argument of `asdf.open()` and `AsdfFile` is deprecated,

>     and will be removed in ASDF 4.0. It is replaced by the more-explicitly-named
>     ``memmap`` argument as the inverse of ``copy_arrays``
>     (``memmap == not copy_arrays``). Using ``copy_arrays`` will work for now, but
>     give an ``AsdfWarning``.

Author: ```suggestion
    and will be removed in ASDF 4.0. It is replaced by ``memmap``, which
    is the opposite of ``copy_arrays`` (``memmap == not copy_arrays``).
    In ASDF 4.0, ``memmap`` will default to ``False``, which means arrays 
    will no longer be memory-mapped by default.
```

    >>> np.equal(af["powers"]["squares"], expected).all()
    True

> By default, uncompressed data blocks are not memory mapped.
> Memory mapping can be enabled for efficient access by using
> the ``memmap`` option of `open` when reading:

Reviewer: ```suggestion
By default, uncompressed data blocks are not memory mapped for efficient
access. Memory mapping can be disabled by using the ``memmap``
option of `open` when reading:
```

.. code:: python

>     af = asdf.open("example.asdf", memmap=True)

Reviewer: ```suggestion
    af = asdf.open("example.asdf", memmap=False)
```
